packages/transformer/src/IModelTransformer.ts

```diff
@@ -411,18 +414,144 @@ export class IModelTransformer extends IModelExportHandler {
+        : undefined;
+    });
   }
 
-  private get _isForwardSynchronization() {
-    return this._isSynchronization && !this._options.isReverseSynchronization;
+  /**
+   * Determines the sync direction "forward" or "reverse" of a given sourceDb and targetDb by looking for the scoping ESA.
+   * If the sourceDb's iModelId is found as the identifier of the expected scoping ESA in the targetDb, then it is a forward synchronization.
+   * If the targetDb's iModelId is found as the identifier of the expected scoping ESA in the sourceDb, then it is a reverse synchronization.
+   * @throws if no scoping ESA can be found in either the sourceDb or targetDb which describes a master branch relationship between the two databases.
+   * @param sourceDb
+   * @param targetDb
+   * @param targetScopeElementId the element Id defined on the ExternalSourceAspect's scope. @see [[ExternalSourceAspectProps]]
+   * @returns "forward" or "reverse"
+   */
+  public static determineSyncType(
+    sourceDb: IModelDb,
+    targetDb: IModelDb,
+    targetScopeElementId: Id64String
+  ): "forward" | "reverse" {
+    const aspectProps = {
+      id: undefined as string | undefined,
+      version: undefined as string | undefined,
+      classFullName: ExternalSourceAspect.classFullName,
+      element: {
+        id: targetScopeElementId,
+        relClassName: ElementOwnsExternalSourceAspects.classFullName,
+      },
+      scope: { id: IModel.rootSubjectId }, // the root Subject scopes scope elements
+      identifier: sourceDb.iModelId,
+      kind: ExternalSourceAspect.Kind.Scope,
+      jsonProperties: undefined as TargetScopeProvenanceJsonProps | undefined,
+    };
+    // First assume that the targetDb is the provenanceDb/branch (meaning holds the provenance which points back to master)
+    this.queryScopeExternalSource(targetDb, aspectProps, {
+      getJsonProperties: true,
+    });
+    if (undefined !== aspectProps.id) {
+      return "forward"; // we found an esa assuming targetDb is the provenanceDb/branch so this is a forward sync.
+    }
+
+    // Now assume that the sourceDb is the provenanceDb/branch
+    aspectProps.identifier = targetDb.iModelId;
+    this.queryScopeExternalSource(sourceDb, aspectProps, {
+      getJsonProperties: true,
+    });
+
+    if (undefined !== aspectProps.id) {
+      return "reverse"; // we found an esa assuming sourceDb is the provenanceDb/branch so this is a reverse sync.
+    }
+    throw new Error(this.noEsaSyncDirectionErrorMessage);
+  }
+
+  private determineSyncType(): SyncType {
+    if (this._isProvenanceInitTransform) {
+      return "forward";
+    }
+    if (!this._isSynchronization) {
+      return "not-sync";
+    }
+    try {
+      return IModelTransformer.determineSyncType(
+        this.sourceDb,
+        this.targetDb,
+        this.targetScopeElementId
+      );
+    } catch (err) {
+      if (
+        err instanceof Error &&
+        err.message === IModelTransformer.noEsaSyncDirectionErrorMessage &&
```

Comment:
nit, consider:
```ts
interface SyncDirectionErr extends Error {
  code: "TRANSFORMER-SYNCDIR";
}
const isSyncDirectionErr = (e: Error) => (e as SyncDirectionErr).code === "TRANSFORMER-SYNCDIR";
```
or
```ts
class SyncDirectionError extends Error {}
// and use `instanceof`
```

Note that I consider `instanceof` to be bad practice in javascript because it isn't duck typed and doesn't work if you end up importing multiple versions of one library. Not a major issue but I prefer to not use it in general.